### PR TITLE
fix: issue 4 in #56

### PR DESCRIPTION
**Pull request by ramprasadm-CBT**
Ticket #56:

QA History:
-> QA notified of the issue in organization page. "Getting yellow page issue on ADD location page, when we add exceed number in phone number , it should a toast message". On investigation the issue is related to permission of different groups in the database.

Steps Taken:
-> Added a Groupmixin in both class based and function based views of organization.
-> Added a custom 403 page.
image

other issue fixed:
-> Some pages in tenant login displays yellow page issue on load. Fixed confirmed and cancelled organization list page on the tenant side.
-> Fixed Issue 3 in ticket#51.

### CODE REVIEW

File: backend/base/templates/cancelled_organization.html [hunk 1]
```diff
@@ -1,250 +1,167 @@
...
-        </footer>
-      </div>
+          <a href="{% url 'tenantuser_page' %}" class="btn btn-primary mt-4" data-toggle="tooltip" title="Back" style="border-radius: 50%; height: max-content; width: max-content; margin-left: auto; margin-right: auto; display: block;">
+            <i class="fas fa-arrow-left"></i>
+          </a>
+        </div>
+      </main>
+      <footer class="py-4 bg-light mt-auto">
+        <div class="container-fluid px-4 d-flex align-items-center justify-content-center">
+          <div class="text-muted">Copyright &copy; Your Website 2023</div>
```
Comment (murucbt) [inline backend/base/templates/cancelled_organization.html]: Still in 2023?

File: backend/base/views.py [hunk 2]
```diff
@@ -282,8 +310,9 @@ def get(self, request):
         return redirect('login')
 
 @method_decorator(login_required, name='dispatch')
-class OrganizationHomeView(TemplateView):
+class OrganizationHomeView(GroupAccessMixin,TemplateView):
```
Comment (murucbt) [inline backend/base/views.py:313]: give space after comma(, )

File: backend/base/views.py [hunk 3]
```diff
@@ -305,8 +334,9 @@ def get_context_data(self, **kwargs):
         return context
 
 @method_decorator(login_required, name='dispatch')
-class ListofConfirmBookingView(TemplateView):
+class ListofConfirmBookingView(GroupAccessMixin,TemplateView):
```
Comment (murucbt) [inline backend/base/views.py]: same comment here too and below wherever you used `GroupAccessMixin`.

File: backend/base/templates/cancelled_organization.html [hunk 4]
```diff
@@ -1,250 +1,167 @@
...
+      const sidebarToggle = document.querySelector("#sidebarToggle");
+      if (sidebarToggle) {
+        sidebarToggle.addEventListener("click", (event) => {
+          event.preventDefault();
+          document.body.classList.toggle("sb-sidenav-toggled");
+          localStorage.setItem(
+            "sb|sidebar-toggle",
+            document.body.classList.contains("sb-sidenav-toggled")
+          );
+        });
```
Comment (murucbt) [inline backend/base/templates/cancelled_organization.html]: Page has complete indentation issues. Mixed with space and tabs. Please be sure to sure anyone.

File: backend/base/templates/cancelled_organization.html [hunk 5]
```diff
@@ -1,250 +1,167 @@
...
+        </div>
+        <div class="sb-sidenav-footer">
+          <div class="small">Logged in as:</div>
+          <div class="user-email">{{ request.user.email }}</div>
+        </div>
+      </nav>
+    </div>
+    
+    <div id="layoutSidenav_content">
+      <main>
```
Comment (murucbt) [inline backend/base/templates/cancelled_organization.html:87]: what is the use of <main> tag?

File: backend/base/templates/cancelled_organization.html [hunk 6]
```diff
@@ -1,250 +1,167 @@
...
+  <meta name="author" content="" />
+  <title>STRONGR.</title>
+  <link href="https://cdn.jsdelivr.net/npm/simple-datatables@7.1.2/dist/style.min.css" rel="stylesheet" />
+  <script src="https://use.fontawesome.com/releases/v6.3.0/js/all.js" crossorigin="anonymous"></script>
+  <link href="../static/styles.css" rel="stylesheet" />
+  <link href="{% static './css/org-status-list.css' %}" rel="stylesheet" />
+  <link rel="preconnect" href="https://fonts.googleapis.com" />
+  <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin />
+  <link href="https://fonts.googleapis.com/css2?family=Poppins:wght@400;700&display=swap" rel="stylesheet" />
+</head>
```
Comment (murucbt) [inline backend/base/templates/cancelled_organization.html:19]: Are we still using header for each page?

File: backend/base/templates/403.html [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+          </g>
+        </svg>
+<h1>You are not allowed to enter here</h1>
+<h2>Go <a target="_blank" href="#">Home!</a></h2>
+
+<script>
+  var root = document.documentElement;
+var eyef = document.getElementById('eyef');
+var cx = document.getElementById("eyef").getAttribute("cx");
+var cy = document.getElementById("eyef").getAttribute("cy");
```
Comment (murucbt) [inline backend/base/templates/403.html]: space tab issue here. why `var`? can't this be `const`?

File: backend/base/templates/403.html [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+
+document.addEventListener("mousemove", evt => {
+  let x = evt.clientX / innerWidth;
+  let y = evt.clientY / innerHeight;
```
Comment (murucbt) [inline backend/base/templates/403.html]: Why let ? can't you use const when you don't reassign the value of x or y here?

File: backend/base/static/css/403.css [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+  cy: calc(50px + 30px * var(--mouse-y));
+}
+#eye-wrap {
+  overflow: hidden;
+}
+.error-text {
+  font-size: 120px;
+}
+.alarm {
+  animation: alarmOn 0.5s infinite;
```
Comment (murucbt) [inline backend/base/static/css/403.css:42]: what is alarmOn?

File: backend/base/templates/403.html [hunk 10]
```diff
@@ -0,0 +1,56 @@
...
+  cx = 115 + 30 * x;
+  cy = 50 + 30 * y;
+  eyef.setAttribute("cx", cx);
+  eyef.setAttribute("cy", cy);
+  
+});
+
+document.addEventListener("touchmove", touchHandler => {
+  let x = touchHandler.touches[0].clientX / innerWidth;
+  let y = touchHandler.touches[0].clientY / innerHeight;
```
Comment (murucbt) [inline backend/base/templates/403.html]: same comment as above! why let?

File: backend/base/templates/cancelled_organization.html [hunk 11]
```diff
@@ -1,250 +1,167 @@
...
-      </div>
-      <div id="layoutSidenav_content">
-        <main>
-          <div class="container-fluid px-4">
-            <h1 class="header1 mt-4 mb-4" style="margin-left: 30%">
-              Cancelled Organization List
-            </h1>
+<body class="sb-nav-fixed">
+  <nav class="sb-topnav navbar navbar-expand navbar-dark bg-dark">
+    <a class="navbar-brand ps-3" href="{% url 'tenantuser_page' %}">STRONGR.</a>
```
Comment (murucbt) [inline backend/base/templates/cancelled_organization.html:23]: Don't we have constant for STRONGR. somewhere to use it in template via view or directly in. HTML?

**Review by murucbt — CHANGES_REQUESTED** [review]
Space & Tab issue and some more comments to address and fix.

File: backend/base/templates/cancelled_organization.html (hunk 1 above)
Comment (ramprasadm-CBT) [reply backend/base/templates/cancelled_organization.html]: Will change now. I was focusing on yellow page issue caused due to indentation problems.

File: backend/base/templates/cancelled_organization.html (hunk 11 above)
Comment (ramprasadm-CBT) [reply backend/base/templates/cancelled_organization.html:23]: Used in React Pages. Will change HTML pages too now

File: backend/base/templates/cancelled_organization.html (hunk 6 above)
Comment (ramprasadm-CBT) [reply backend/base/templates/cancelled_organization.html:19]: base page is used for organization pages only. Will add it for Tenant pages now

File: backend/base/templates/cancelled_organization.html (hunk 5 above)
Comment (ramprasadm-CBT) [reply backend/base/templates/cancelled_organization.html:87]: Which tag anna?

File: backend/base/static/css/403.css (hunk 9 above)
Comment (ramprasadm-CBT) [reply backend/base/static/css/403.css:42]: Its for red light swirling effect on the top of 0.  A self defined function for adding effect via keyframes

File: backend/base/templates/403.html (hunk 8 above)
Comment (ramprasadm-CBT) [reply backend/base/templates/403.html]: will change to const now,

**Review by TharunCBT — APPROVED** [review]

**Pull request merged**